pallets/runtime/tests/src/relayer_test.rs

```diff
@@ -203,6 +202,56 @@ fn do_basic_relayer_paying_key_test() {
     );
 }
 
+#[test]
+fn update_polyx_limit_test() {
+    ExtBuilder::default()
+        .monied(true)
+        .build()
+        .execute_with(&do_update_polyx_limit_test);
+}
+fn do_update_polyx_limit_test() {
+    let bob = User::new(AccountKeyring::Bob);
+    let alice = User::new(AccountKeyring::Alice);
+
+    let set = |p: User, x| Relayer::update_polyx_limit(p.origin(), bob.acc(), x);
+    let add = |p: User, x| Relayer::increase_polyx_limit(p.origin(), bob.acc(), x);
+    let sub = |p: User, x| Relayer::decrease_polyx_limit(p.origin(), bob.acc(), x);
+
+    let test_update = |res, expected_res: Result<(), Error>, expected_limit| {
+        if let Err(e) = expected_res {
+            assert_noop!(res, e);
```

Comment:
This doesn't work as intended unfortunately. The macro `assert_noop!` is defined like so:
```rust

macro_rules! assert_noop {
	(
		$x:expr,
		$y:expr $(,)?
	) => {
		let h = $crate::storage_root();
		$crate::assert_err!($x, $y);
		assert_eq!(h, $crate::storage_root());
	}
}
```
As you can see, it wraps the execution of the expression `$x` in first a statement to store the current storage root and then makes sure that the storage root after hasn't changed. However, you've already done all storage changes outside the call to `test_update(...)` by doing `add(bob, 100u128)`, since Rust has eager evaluation semantics.

To fix this, you have two options (if we had generic closures you would have had 3):
1. Make `test_update` a macro.
2. Follow my original suggestion more exactly with the `for` loop.